graph/pull_v2.go

```diff
@@ -74,14 +74,22 @@ func (p *v2Puller) pullV2Repository(tag string) (err error) {
 	}
 
 	broadcaster, found := p.poolAdd("pull", taggedName)
+	broadcaster.Add(p.config.OutStream)
 	if found {
 		// Another pull of the same repository is already taking place; just wait for it to finish
-		broadcaster.Add(p.config.OutStream)
-		broadcaster.Wait()
+		result := broadcaster.Wait()
+		if err, ok := result.(error); ok {
+			return err
+		}
 		return nil
 	}
-	defer p.poolRemove("pull", taggedName)
-	broadcaster.Add(p.config.OutStream)
+	defer func() {
+		if err != nil {
```

Comment:
I believe that if this block was replaced with `p.poolRemove("pull", taggedName, err)`, then in the case where `err` is nil, the type switch cases that check for an `error` would succeed. In other words, a nil error is different than a nil `interface{}`. I'd love suggestions of a cleaner way to deal with this. I agree it's ugly.